pkg/api/componentreadiness/component_report_test.go

```diff
@@ -1496,16 +1524,124 @@ func Test_componentReportGenerator_assessComponentStatus(t *testing.T) {
+			baseFlake:              1,
+			numberOfIgnoredSamples: 9,
+			expectedStatus:         -5,
+			expectedFischers:       thrift.Float64Ptr(0.285714285714284),
+		},
+		{
+			name:                        "new test no regression",
+			sampleTotal:                 1000,
+			sampleSuccess:               999,
+			requiredPassRateForNewTests: 99,
+			expectedStatus:              crtype.MissingBasis,
+			expectedFischers:            nil,
+		},
+		{
+			name:                        "new test extreme regression",
+			sampleTotal:                 15,
+			sampleSuccess:               13,
+			requiredPassRateForNewTests: 99,
+			expectedStatus:              crtype.ExtremeRegression,
+			expectedFischers:            nil,
+		},
+		{
+			name:                        "new test significant regression",
+			sampleTotal:                 1000,
+			sampleSuccess:               985,
+			requiredPassRateForNewTests: 99,
+			expectedStatus:              crtype.SignificantRegression,
+		},
+		{
+			name:                        "new test significant regression with triaged runs",
+			sampleTotal:                 1000,
+			sampleSuccess:               985,
+			numberOfIgnoredSamples:      12,
+			requiredPassRateForNewTests: 99,
+			expectedStatus:              crtype.MissingBasis,
+		},
+		{
+			name:                        "new test all failures triaged",
+			sampleTotal:                 1000,
+			sampleSuccess:               988,
+			numberOfIgnoredSamples:      12,
+			requiredPassRateForNewTests: 99,
+			expectedStatus:              crtype.MissingBasis,
+		},
+		{
+			name:                        "pass rate mode significant regression",
+			sampleTotal:                 100,
+			sampleSuccess:               94,
+			sampleFlake:                 0,
+			baseTotal:                   100,
+			baseSuccess:                 94,
+			baseFlake:                   0,
+			numberOfIgnoredSamples:      0,
+			requiredPassRateForAllTests: 95,
+			expectedStatus:              crtype.SignificantRegression,
+		},
+		{
+			name:                        "pass rate mode extreme regression",
+			sampleTotal:                 100,
+			sampleSuccess:               70,
+			sampleFlake:                 0,
+			baseTotal:                   100,
+			baseSuccess:                 70,
+			baseFlake:                   0,
+			numberOfIgnoredSamples:      0,
+			requiredPassRateForAllTests: 95,
+			expectedStatus:              crtype.ExtremeRegression,
+		},
+		{
+			name:                        "pass rate mode no regression",
+			sampleTotal:                 100,
+			sampleSuccess:               97,
+			sampleFlake:                 0,
+			baseTotal:                   100,
+			baseSuccess:                 97,
+			baseFlake:                   0,
+			numberOfIgnoredSamples:      0,
+			requiredPassRateForAllTests: 95,
+			expectedStatus:              crtype.NotSignificant,
+		},
+		{
+			name:                        "pass rate mode significant regression ignores minimum failures",
```

Comment:
Note I do not consider min failures here, it felt contradictory to pass rate testing. If you've got a new test and it's only run 7x in a week, and one failed, that should not meet the bar.